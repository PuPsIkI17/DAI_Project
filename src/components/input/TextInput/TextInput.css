.input {
    font-size: 16px;
    padding: 0;
    margin: 0;
}

.input>label {
    display: inline-block;
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    font-size: 16px;
    line-height: 1.5;
    color: #333;
}

.input>input {
    font-family: "Roboto", sans-serif;
    background: #f2f2f2;
    width: 100%;
    margin: 0 0 5px;
    padding: 10px;
    box-sizing: border-box;
    font-size: 16px;
    border: 1px solid #285e8e;
    border-radius: 5px;
}