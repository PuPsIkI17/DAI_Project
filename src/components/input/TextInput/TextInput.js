import React from 'react';
import PropTypes from 'prop-types';
import './TextInput.css'
export default class TextInput extends React.Component {
    render() {
        const options = {};
        this.props.value !== undefined ? options["value"] = this.props.value : options["defaultValue"] = this.props.defaultValue;
        if (this.props.min) {
            options['min'] = this.props.min
        }
        return (
            <div className="input">
                <label htmlFor={this.props.inputId}>{this.props.label}</label>
                <input
                    type={this.props.type}
                    placeholder={this.props.placeholder}
                    name={this.props.inputId}
                    id={this.props.id}
                    {...options}
                    required={this.props.required}
                    onChange={this.props.onChange} />

            </div>
        )
    }
}
TextInput.propTypes = {
    label: PropTypes.string,
    id: PropTypes.string,
    type: PropTypes.string,
    placeholder: PropTypes.string
}
TextInput.defaultProps = {
    label: 'Label',
    id: 'defaultInput',
    type: 'text',
    placeholder: '',
    defaultValue: '',
    required: false,
    onChange: () => { }
}