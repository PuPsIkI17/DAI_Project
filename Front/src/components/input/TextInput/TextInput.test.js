import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';
describe('TextInput Component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TextInput />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const input = renderer
            .create(<TextInput />)
            .toJSON();
        expect(input).toMatchSnapshot();
    });
})
