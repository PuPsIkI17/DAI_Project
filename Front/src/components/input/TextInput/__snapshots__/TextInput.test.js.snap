// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component renders the UI as expected 1`] = `
<div
  className="input"
>
  <label>
    Label
  </label>
  <input
    defaultValue=""
    id="defaultInput"
    onChange={[Function]}
    placeholder=""
    required={false}
    type="text"
  />
</div>
`;
